pkg/hubble/parser/seven/http.go

```diff
@@ -24,7 +35,15 @@ func decodeHTTP(flowType accesslog.FlowType, http *accesslog.LogRecordHTTP, opts
 	sort.Strings(keys)
 	for _, key := range keys {
 		for _, value := range http.Headers[key] {
-			headers = append(headers, &flowpb.HTTPHeader{Key: key, Value: value})
+			if opts.RedactHTTPHeaders {
```

Comment:
A structure like
```
if opts.RedactHTTPHeaders && !slices.Contains(allowedHeaders, strings.ToLower(key)) {
    value = defaults.SensitiveValueRedacted
}
headers = append(headers, &flowpb.HTTPHeader{Key: key, Value: value})
```
would be more readable I think.

Also, instead of checking `slices.Contains` every time, I would represent the allow list as a `map[string]struct{}` and check headers like `if _, ok := set[header]; ok ...`, see [here](https://emersion.fr/blog/2017/sets-in-go/). I'm not sure if compiler would similarly optimize this check?